### PR TITLE
ERC-6123: Move to Draft: Enlarged interface, streamlined reference implementation.

### PR DESCRIPTION
**Pull request by cfries**
This pull request will make some additions to the ERC-6123 interface, streamlined the reference implementation and improved the documentation.
Run unit test via npm install; npm test

### CODE REVIEW

**Comment by SamWilsn** [comment]
I *think* if you rebase this, it'll pass the link checker.

**Comment by cfries** [comment]
> I _think_ if you rebase this, it'll pass the link checker.

There appear to be link errors in other ERCs - not in this one:

```
For the Images check, the following failures were found:
* At ./_site/EIPS/eip-6358.html:226:
  internal image ../assets/erc-6358/img/o-dlt.png does not exist
* At ./_site/EIPS/eip-7439.html:365:
  internal image ../assets/erc-7439/normal.png does not exist
* At ./_site/EIPS/eip-7439.html:369:
  internal image ../assets/erc-7439/void.png does not exist```

I proposed to fix them here: https://github.com/ethereum/ERCs/pull/28 - but I assume it was decided to use another aproach. 

How can I move this EIP towards being merged? Its open since Oct 26. If I can be of any help, let me know.

**Comment by cfries** [comment]
> I _think_ if you rebase this, it'll pass the link checker.

Thank you. Looks good now.

File: ERCS/erc-6123.md [hunk 1]
```diff
@@ -125,101 +142,97 @@ event TradeConfirmed(address confirmer, string tradeId);
...
-A reference implementation SDC.sol is provided and is based on the [EIP-20](./eip-20.md) token standard. 
-See folder /assets/contracts, more explanation on the implementation is provided inline.
+An abstract contract class SDC.sol as well as a full reference implementation SDCPledgedBalance.sol for an OTC-Derivative is provided and is based on the [ERC-20](./eip-20.md) token standard.
+See folder `/assets/contracts`, more explanation on the implementation is provided inline.
+
+### NPM Packages
+
+The current definition of the SDC interface is available via `@finmath.net/sdc@0.3.0`.
+
+Note: The previous draft version of interface (used in the 2022 test trade) is `@finmath.net/sdc@0.2.3`.
```
[the remark this answers is not in the document]
Comment (cfries) [reply ERCS/erc-6123.md]: Done.

File: ERCS/erc-6123.md [hunk 2]
```diff
@@ -1,111 +1,128 @@
...
 
 ## Specification
 
 ### Methods
 
-The following methods specify  inception and post-trade live cycle of a Smart Derivative Contract. For futher information also please look at the interface documentation ISDC.sol.
+The following methods specify a Smart Derivative Contract's trade initiation and settlement life cycle. For further information, please also look at the interface documentation ISDC.sol.
 
-#### inceptTrade
+#### Trade Initiation Phase: inceptTrade
```
Comment (SamWilsn) [inline ERCS/erc-6123.md:63]:
```suggestion
#### Trade Initiation Phase: `inceptTrade`
```

File: ERCS/erc-6123.md [hunk 3]
```diff
@@ -1,111 +1,128 @@
...
 ```
 
-#### requestTermination
 
-Allows an eligible party to request a mutual termination
+#### Trade Termination: `requestTermination`
+
+Allows an eligible party to request a mutual termination with a termination amount she is willing to pay
 
 ```js
```
Comment (SamWilsn) [inline ERCS/erc-6123.md:111]:
````suggestion
```solidity
````

File: ERCS/erc-6123.md [hunk 4]
```diff
@@ -125,101 +142,91 @@ event TradeConfirmed(address confirmer, string tradeId);
...
+- The provided interface specification is supposed to completely reflect the entire trade life cycle.
+- The interface specification is generic enough to handle the case that parties process one or even multiple financial transactions (on a netted base)
+- Usually, the valuation of financial trades (e.g. OTC Derivatives) will require advanced valuation methodology to determine the market value. This is why the concept might rely on an external market data source and hosted valuation algorithms
+- A pull-based valuation-based oracle pattern can be implemented by using the provided callback pattern (methods: `initiateSettlement`, `performSettlement`)
 - The reference implementation `SDC.sol` is based on a state-machine pattern where the states also serve as guards (via modifiers) to check which method is allowed to be called at a particular given process and trade state
-- Java based state machine and contract implementations are also available. See the github repo link below.
+- Java-based state machine and contract implementations are also available. See the finmath-smart-derivative-contract GitHub repo..
```
Comment (SamWilsn) [inline ERCS/erc-6123.md:211]: Please remove this external link.

```suggestion
```

**Review by SamWilsn — CHANGES_REQUESTED** [review]
I missed this external link on my last review. Sorry about that. You'll need to remove it, and any others, before we can merge.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**